### PR TITLE
refactor: split ProcessBotDelegatedWithdrawals

### PR DESCRIPTION
**Pull request by bendanzhentan**
Resolve https://github.com/bnb-chain/opbnb-bridge-bot/pull/1#discussion_r1400296694

### CODE REVIEW

File: cmd/bot/run.go [hunk 1]
```diff
@@ -76,65 +76,80 @@ func runCommand(ctx *cli.Context) error {
...
-	limit := 1000
-
 	for {
 		select {
 		case <-ticker.C:
 		case <-ctx.Done():
 			return
+		default:
+			ProcessUnprovenBotDelegatedWithdrawals(ctx, log, db, l1Client, l2Client, cfg)
+			ProcessUnfinalizedBotDelegatedWithdrawals(ctx, log, db, l1Client, l2Client, cfg)
```
Comment (owen-reorg) [inline cmd/bot/run.go:83]: should be in the `case <- ticker.C` instead of `default`

Comment (bendanzhentan) [reply cmd/bot/run.go:83]: Thanks for pointing out this bug.
I fix it at [c5fcd17](https://github.com/bnb-chain/opbnb-bridge-bot/pull/7/commits/c5fcd17cf872dbea790ee802283b7e0bee5792b2), please take a look.

File: cmd/bot/run.go [hunk 2]
```diff
@@ -76,65 +76,79 @@ func runCommand(ctx *cli.Context) error {
...
+		err := processor.FinalizeMessage(ctx, &unfinalized)
+		if err != nil {
+			if strings.Contains(err.Error(), "OptimismPortal: withdrawal has already been finalized") {
+				// The withdrawal has already finalized, mark it
+				result := db.Model(&unfinalized).Update("finalized", true)
+				if result.Error != nil {
+					log.Error("failed to update finalized l2_contract_events", "error", result.Error)
+					return
 				}
+			} else if strings.Contains(err.Error(), "OptimismPortal: withdrawal has not been proven yet") || strings.Contains(err.Error(), "OptimismPortal: proven withdrawal finalization period has not elapsed") {
```
Comment (welkin22) [inline cmd/bot/run.go]: `OptimismPortal: withdrawal has not been proven yet` appeared, but all the data we queried before were `proven = true`, which indicates that there is a problem with the indexer. I think there should be an error log here.

Comment (bendanzhentan) [reply cmd/bot/run.go]: Make sense.
Updated at [6df5778](https://github.com/bnb-chain/opbnb-bridge-bot/pull/7/commits/6df57789b0162300332975e491c9434313f6beb9), @welkin22 PTAL

**Review by welkin22 — APPROVED** [review]
LGTM

**Review by owen-reorg — APPROVED** [review]

**Pull request merged**